crates/cast/bin/cmd/create2.rs

```diff
@@ -114,34 +124,62 @@ impl Create2Args {
             keccak256(hex::decode(init_code)?)
         };
 
+        let mut n_threads = std::thread::available_parallelism().map_or(1, |n| n.get());
+        if let Some(jobs) = jobs {
+            n_threads = n_threads.min(jobs);
+        }
+        if cfg!(test) {
+            n_threads = n_threads.min(2);
+        }
+
         println!("Starting to generate deterministic contract address...");
+        let mut handles = Vec::with_capacity(n_threads);
+        let found = Arc::new(AtomicBool::new(false));
         let timer = Instant::now();
-        let (salt, addr) = std::iter::repeat(())
-            .par_bridge()
-            .map(|_| {
-                let salt = B256::random();
-
-                let addr = deployer.create2(salt, init_code_hash).to_checksum(None);
-
-                (salt, addr)
-            })
-            .find_any(move |(_, addr)| {
-                let addr = addr.to_string();
-                let addr = addr.strip_prefix("0x").unwrap();
-                regex.matches(addr).into_iter().count() == regex.patterns().len()
-            })
-            .unwrap();
-
-        let salt = U256::from_be_bytes(*salt);
-        let address = Address::from_str(&addr).unwrap();
-
-        println!(
-            "Successfully found contract address in {} seconds.\nAddress: {}\nSalt: {}",
-            timer.elapsed().as_secs(),
-            addr,
-            salt
-        );
+        for i in 0..n_threads {
+            let increment = n_threads;
+            let start = i;
+            let regex = regex.clone();
+            let regex_len = regex.patterns().len();
+            let found = Arc::clone(&found);
+            handles.push(std::thread::spawn(move || {
+                struct B256Aligned(B256, [usize; 0]);
+                let mut salt = B256Aligned(B256::ZERO, []);
+                let salt_word = unsafe { &mut *salt.0.as_mut_ptr().cast::<usize>() };
+                *salt_word = start;
```

Comment:
not obvious what's going on here, this sets the salt's first word to the thread's index incrementing it by the number of threads, so the salt cycles through all unique values?

needs docs+SAFETY

Reply:
n_threads=4
- word=0; word=4; word=8
- word=1; word=5; word=9
- word=2; word=6; word=10
- word=3; word=7; word=11